src/sentry/models/grouphash.py

```diff
@@ -38,36 +38,22 @@ class Meta:
         unique_together = (('project', 'hash'), )
 
     @staticmethod
-    def fetch_last_processed_event_id(project_id, group_hash_ids):
-        prefix = 'last-processed-event:{}'.format(project_id)
+    def fetch_last_processed_event_id(group_id, group_hash_ids):
         with redis.clusters.get('default').map() as client:
-            results = map(
-                lambda group_hash_id: client.hget(
-                    '{}:{}'.format(prefix, group_hash_id % 16),
-                    group_hash_id,
-                ),
-                group_hash_ids,
+            result = client.hmget(
+                'gh:lp:{}'.format(group_id),
+                map('{}'.format, group_hash_ids),
             )
 
-        return map(
-            lambda result: result.value,
-            results,
-        )
+        return result.value
 
     @staticmethod
-    def record_last_processed_event_id(project_id, group_hash_ids, event_id):
-        prefix = 'last-processed-event:{}'.format(project_id)
+    def record_last_processed_event_id(group_id, group_hash_id, event_id):
```

Comment:
Just remembered that the group hash group is mutable which makes this a little bit more difficult. 😓 

Going to move this to a top level key instead, I guess.